proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/ProxySQLExecutor.java

```diff
@@ -143,16 +134,40 @@ private boolean isPostgreSQLOrOpenGaussStatement(final SQLStatement sqlStatement
      * @throws SQLException SQL exception
      */
     public List<ExecuteResult> execute(final ExecutionContext executionContext) throws SQLException {
+        return needAutoTransaction(executionContext) ? doExecuteWithinTransaction(executionContext) : doExecute(executionContext);
+    }
+    
+    private boolean needAutoTransaction(final ExecutionContext executionContext) {
+        TransactionStatus transactionStatus = backendConnection.getConnectionSession().getTransactionStatus();
+        SQLStatement sqlStatement = executionContext.getSqlStatementContext().getSqlStatement();
+        return TransactionType.XA == transactionStatus.getTransactionType() && !transactionStatus.isInTransaction() && sqlStatement instanceof DMLStatement
+                && !(sqlStatement instanceof SelectStatement) && executionContext.getExecutionUnits().size() > 1;
+    }
+    
+    private List<ExecuteResult> doExecuteWithinTransaction(final ExecutionContext executionContext) throws SQLException {
+        List<ExecuteResult> result;
+        JDBCBackendTransactionManager transactionManager = new JDBCBackendTransactionManager(backendConnection);
```

Comment:
Is it possible to avoid creating the JDBCBackendTransactionManager object every time?

Reply:
The `JDBCBackendTransactionManager` just only holds the `ShardingSphereTransactionManagerEngine` object, which has been created in the rule in advance.